### PR TITLE
Add fallback in requirements check, only check once

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

The requirements check should only warn and never raise an error, so add a fallback for any requirements that can't be handled otherwise. In addition, add a flag to `project_run` so that the requirements check is only run once for the top-level `run` command rather than also for each subcommand in workflows.

### Types of change

Bug fix.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
It seems like it would be nicer to override `check_requirements` in the config, but this isn't a supported override.

I'm also not attached to the `skip_check_requirements` naming. `skip_requirements_check` might be better?

File: spacy/cli/project/run.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def project_run(
     force: bool = False,
     dry: bool = False,
     capture: bool = False,
+    skip_check_requirements: bool = False,
```
Comment (rmitsch) [inline spacy/cli/project/run.py:56]: I favor `skip_requirements_check` over `skip_check_requirements`.

**Review by rmitsch — APPROVED** [review]

**Comment by adrianeboyd** [comment]
You get:

```
check_requirements	not a section value that can be overwritten
```

It seems to look for `.` in any overrides.

File: spacy/cli/project/run.py [hunk 2]
```diff
@@ -69,16 +70,18 @@ def project_run(
...
     validate_subcommand(list(commands.keys()), list(workflows.keys()), subcommand)
 
     req_path = project_dir / "requirements.txt"
-    if config.get("check_requirements", True) and os.path.exists(req_path):
-        with req_path.open() as requirements_file:
-            _check_requirements([req.replace("\n", "") for req in requirements_file])
+    if not skip_requirements_check:
+        if config.get("check_requirements", True) and os.path.exists(req_path):
+            with req_path.open() as requirements_file:
+                _check_requirements([req.replace("\n", "") for req in requirements_file])
```
Comment (polm) [inline spacy/cli/project/run.py:84]:
```suggestion
                _check_requirements([req.strip() for req in requirements_file])
```

I had to check, but this is more like what `pip install -r` does. In general this is not important, but the current version won't work with DOS newlines, for example.

**Review by polm — COMMENTED** [review]
There's one minor thing I think should be fixed but looks good.

**Comment by polm** [comment]
Doesn't this require a docs update?

**Comment by adrianeboyd** [comment]
I think that v3.4.2 should have had the relevant doc updates (#11226)? Unless you mean the `@` part?

**Comment by polm** [comment]
Ah, yes, that should be fine - I forgot this was a followup to that PR.

**Review by thomashacker — APPROVED** [review]

**Pull request merged**